urllib3/connection.py

```diff
@@ -299,12 +302,26 @@ def connect(self):
                     'for details.)'.format(hostname)),
                     SubjectAltNameWarning
                 )
-            match_hostname(cert, self.assert_hostname or hostname)
+            _match_hostname(cert, self.assert_hostname or hostname)
 
         self.is_verified = (resolved_cert_reqs == ssl.CERT_REQUIRED or
                             self.assert_fingerprint is not None)
 
 
+def _match_hostname(cert, asserted_hostname):
```

Comment:
I mean this doesn't hurt, but I didn't mean that this *had* to happen. I think we were talking about monkey-patching as if someone was going to reuse this code (which I don't think anyone will).  If we are expecting people to reuse this, this should be a function that is not faux-private living in `urllib3.util._ssl`.

This is fine now but just seems like unnecessary refactoring.